frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -426,15 +426,18 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
 
       if (_.isEmpty(_.compact(_.values(errors)))) {
         k8sCreate(operandModel, k8sObj)
-          .then(() =>
+          .then(() => {
+            const params = new URLSearchParams(window.location.href);
             history.push(
-              `${resourcePathFromModel(
-                ClusterServiceVersionModel,
-                clusterServiceVersion.metadata.name,
-                namespace,
-              )}/${referenceForModel(operandModel)}`,
-            ),
-          )
+              params.has('redirect')
```

Comment:
Why not use perspective extension to get the redirect URL based on perspective similar to how import from git does it right now?